src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -336,11 +352,11 @@ protected AIState attackProtect()
     protected AIState attackPhysical()
     {
 
-        if (worker.getLastAttackedEntity() != null
-              && !worker.getLastAttackedEntity().isDead
-              && worker.getDistance(worker.getLastAttackedEntity()) < getAttackRange())
+        if (worker.getRevengeTarget() != null
+              && !worker.getRevengeTarget().isDead
+              && worker.getDistance(worker.getRevengeTarget()) < getAttackRange())
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)